### PR TITLE
Toast notifications for add item page

### LINKED ISSUE #31: 14. As a user, I want to know if my items were successfully added to the list or not
Users should be able to clearly see if their item was added successfully, and if not, why not.

## Acceptance criteria

- [x] If the item is added successfully, show a confirmation message
- [x] implement notifications with React Hot Toast
#### Show descriptive error messages for the following cases:
- [x] A user tries to submit an empty item
- [x] A user tries to submit an item with no due date specified
- [x] A user tries to add a duplicate item
- [x] A database error occurs

### PR DESCRIPTION
**Pull request by labrocadabro**
Co-authored by @shagunZ 

## Description

Implements toast notifications for error and confirmation messages on the Add Item page.

## Related Issue

closes #31 

## Acceptance Criteria

 - If the item is added successfully, show a confirmation message
 - implement notifications with React Hot Toast
 - 
### Show descriptive error messages for the following cases:
 - A user tries to submit an empty item
 - A user tries to submit an item with no due date specified
 - A user tries to add a duplicate item
 - A database error occurs

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

text notifications

### After

toast notifications

## Testing Steps / QA Criteria

Add an item to the list to see a confirmation message. Attempt to add a duplicate item or an item with no name or no due date to see the errors messages.

### CODE REVIEW

File: src/views/AddItem.jsx [hunk 1]
```diff
@@ -61,26 +59,23 @@ export function AddItem({ data, token }) {
 
 	async function handleSubmit(event) {
```
Comment (RajGM) [inline src/views/AddItem.jsx:60]: There is another way:
toast.promise(
                        ref.set(values),
                        {
                            loading: 'Adding...',
                            success: <b>{values.eventN + " Added"}</b>,
                            error: <b>Error occurred while adding</b>,
                        }
                    );

File: src/App.jsx [hunk 2]
```diff
@@ -48,29 +48,54 @@ export function App() {
...
-					<Route
-						path="/add-item"
-						element={<AddItem data={data} token={listToken} />}
-					/>
-				</Route>
-			</Routes>
-		</Router>
+		<>
+			<Toaster
+				position="bottom-right"
```
Comment (RajGM) [inline src/App.jsx:53]: We avoid in-line css for clean code

**Review by RajGM — APPROVED** [review]

File: src/App.jsx (hunk 2 above)
Comment (labrocadabro) [reply src/App.jsx:53]: That's not actually inline css, it's a prop of the Toaster component.

File: src/views/AddItem.jsx (hunk 1 above)
Comment (labrocadabro) [reply src/views/AddItem.jsx:60]: Thanks, will look into this 👍

**Review by ToobaJamal — APPROVED** [review]

**Pull request merged**